### PR TITLE
Ignore device registration errors

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- [x] I've read and accepted the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- Requires _PATCH_ [version update](https://semver.org/)

### Context
Ignoring device registration errors due to invalid UDID, or Mac devices which are not currently supported.

### CODE REVIEW

File: lib/autoprovision/portal/device_client.rb [hunk 1]
```diff
@@ -43,20 +44,22 @@ def self.ensure_test_devices(test_devices, device_client = Spaceship::Portal.dev
...
         end
 
+        valid_devices = valid_devices.append(test_device)
         raise 'failed to find or create device' unless registered_test_device
       end
 
-      Log.success("every test devices (#{test_devices.length}) registered on bitrise are registered on developer portal")
-      [new_device_registered, portal_devices]
+      Log.success("(#{valid_devices.length}) test devices registered on Bitrise are registered on developer portal")
+      valid_devices
```
Comment (godrei) [inline lib/autoprovision/portal/device_client.rb:62]: It seems here we change the type of the returned array, previously we returned the value of fetch_devices, now we return a list of our test device representation.
Is not it an issue?

Also previously the device list built up: by the devices registered on the developer portal + devices on Bitrise.
Now we only care about the Bitrise test devices.

File: spec/device_client_spec.rb [hunk 2]
```diff
@@ -17,6 +17,27 @@
     allow(fake_portal_client).to receive(:all).and_return(nil)
     allow(fake_portal_client).to receive(:create!).and_return(fake_portal_device)
 
-    Portal::DeviceClient.ensure_test_devices([device], fake_portal_client)
+    valid_devices = Portal::DeviceClient.ensure_test_devices([device], fake_portal_client)
+
+    expect(valid_devices).to eq([device])
+  end
+
+  it 'supresses error due to invalid od mac device UDID' do
```
Comment (godrei) [inline spec/device_client_spec.rb:25]: i suggest extending the test by adding a valid device too, this way we can show that we skip invalid devices, while keep registering valid ones.

**Review by godrei — COMMENTED** [review]
Pls see my notes

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # This configuration was generated by
```
Comment (adborbas) [inline .rubocop_todo.yml:1]: Should .rubocop_todo.yml be reviewed?

Comment (lpusok) [reply .rubocop_todo.yml:1]: no, it is auto-generated to skip rubocop warnings

File: spec/device_client_spec.rb (hunk 2 above)
Comment (lpusok) [reply spec/device_client_spec.rb:25]: Added an existing device, it does not need to be registered tough.

File: lib/autoprovision/portal/device_client.rb [hunk 4]
```diff
@@ -43,20 +44,22 @@ def self.ensure_test_devices(test_devices, device_client = Spaceship::Portal.dev
...
           Log.success("registering test device #{registered_test_device.name} (#{registered_test_device.udid})")
         end
 
+        valid_devices = valid_devices.append(test_device)
         raise 'failed to find or create device' unless registered_test_device
       end
 
-      Log.success("every test devices (#{test_devices.length}) registered on bitrise are registered on developer portal")
-      [new_device_registered, portal_devices]
+      Log.success("(#{valid_devices.length}) test devices registered on Bitrise are registered on developer portal")
```
Comment (godrei) [inline lib/autoprovision/portal/device_client.rb:61]: NOTE:
`3 test devices registered on Bitrise are registered on developer portal` -> `3 test devices registered on Bitrise`

**Review by godrei — APPROVED** [review]

**Review by godrei — APPROVED** [review]

**Pull request merged**